docs/api/patch-envelopes.yml

```diff
@@ -0,0 +1,165 @@
+openapi: 3.0.3
+
+info:
+  title: ISC API
+  description: |
+    Do you want to update your edge applications running on EVE
+    Without need to rebuild edge application? Patch Envelopes API.
+    Check out https://github.com/lf-edge/eve to know what is EVE and Edge computing.
+    Check out https://github.com/lf-edge/eve/blob/master/docs/PATCH-ENVELOPES.md to know more about underlying patch envelopes.
+  version: 0.0.1
+
+servers:
+  - url: http://169.254.169.254/eve/v1
+    description: EVE Metadata endpoint
+
+tags:
+  - name: PatchEnvelopes
+    description: Things related to EVE PatchEnvelopes
+  - name: Status
+    description: Things related to EVE status 
+
+paths:
+  /app/appCustomStatus:
+    post:
+      operationId: postOpaqueStatus
+      summary: Send status to controller.
+      description: |
+        Status is nothing more but a array of bytes send to controller.
+      tags:
+        - Status 
+      responses:
+        '500':
+          description: Something happened during status processing
+        '204':
+          description: Cannot determine app instance id for given IP address
+        '200':
+          description: Successfully sent status
+
+
+  /patch/description.json:
+    get:
+      operationId: getAvailablePatchEnvelopes
+      summary: Returns list of available patch envelopes for edge app instance to download
+      description: |
+        Returned list is in JSON format, returns 422 when there is a problem with
+        processing patch envelopes (should never occur). And 204 when EVE cannot determine
+        App instance ID from given IP address (should also never occur normally)
+      tags:
+        - PatchEnvelopes
+      responses:
+        '200': # JSON message if empty with message
+          description: Patch envelopes available for app instance
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/PatchEnvelopeDescription'
+        '422':
+          description: Something happened during representing patch envelopes for app instance
+        '204':
+          description: Cannot determine app instance id for given IP address
+
+  /patch/download/{patchID}:
+    get:
+      operationId: downloadPatchArchive
+      summary: download patch envelope
+      description: |
+        Archives all artifacts and downloads them
+      tags:
+        - PatchEnvelopes
+      parameters:
+        - in: path
+          name: patchID
+          schema:
+            type: string
+          required: true
+          description: Patch UUID to download
+          example: 69bac069-f4c3-435a-b4e7-d635b60e9b5a
+      responses:
+        '200':
+          description: Returns stream of bytes of archive
```

Comment:
Could we define the body here? From the description I guess the response should contain the body...
It could be useful in the case we later try to add some tools for the implementation against spec verification.

Reply:
@OhmSpectator, I had a tool in separate repo [here](https://github.com/uncleDecart/ics-api/), it's called vacuum and you can use it like [this](https://github.com/uncleDecart/ics-api/blob/319f21f33d17dea5bc9bdacd061def897e46103d/Makefile#L9). Bytes of data are usually enough of description to download this like [here](https://github.com/uncleDecart/ics-container/blob/a3ce1122464a3fad6b06047d261c3bdc516c3d7a/pkg/pemanager/pemanager.go#L182)

Reply:
Ok, cool!
I just remember I used some verification tool that gave me errors in the case when I only set the error codes and didn't set schema: it expected no response body in this case.

Reply:
hmm, this one didn't complain AFAIR, but actually brings a good topic, should we include linter in our makefile. Could be nice to have one, but our Makefile has a lot of things there...

Reply:
You can pick a linter here: https://openapi.tools/
I used Spectral, and it worked well. But I'm not sure it is that critical for us now =)